lib/dry/auto_inject/injector.rb

```diff
@@ -2,22 +2,30 @@
 
 module Dry
   module AutoInject
-    class Injector
+    class Injector < BasicObject
       # @api private
       attr_reader :container
 
       # @api private
       attr_reader :strategy
 
       # @api private
-      def initialize(container, strategy)
+      attr_reader :builder
+
+      # @api private
+      def initialize(container, strategy, builder:)
         @container = container
         @strategy = strategy
+        @builder = builder
       end
 
       def [](*dependency_names)
         strategy.new(container, *dependency_names)
       end
+
+      def method_missing(name, *args, &block)
+        builder.send(name)
```

Comment:
@solnic I keep forgetting about that. Thanks for the reminder – it uses `public_send` now.

@AMHOL `BasicObject` doesn't provide `#respond_to?` _or_ `#respond_to_missing?` so I've implemented a straight-up `#respond_to?` here: https://github.com/dry-rb/dry-auto_inject/pull/25/commits/6db08787abb02afd39d18a72cffdc25742008de3